src/NHibernate/Id/GuidCombGenerator.cs

```diff
@@ -48,20 +50,19 @@ private Guid GenerateComb()
 
 			DateTime now = DateTime.UtcNow;
 
-			// Get the days and milliseconds which will be used to build the byte string 
-			TimeSpan days = new TimeSpan(now.Ticks - BaseDateTicks);
-			TimeSpan msecs = now.TimeOfDay;
+			// We use 2 bytes for the day (65,535 possible values) so we're good until 6/6/2079.
+			var days = new TimeSpan(now.Ticks - BaseDateTicks);
+			var daysArray = BitConverter.GetBytes(days.Days);
 
-			// Convert to a byte array 
-			// Note that SQL Server is accurate to 1/300th of a millisecond so we divide by 3.333333 
```

Comment:
The aim of this is to emulate the SQL Guid Comb algorithm in order to have NHibernate Guid.Comb behaving like its SQL Server expression, `CAST(CAST(NEWID() AS BINARY(10)) 
+ CAST(GETDATE() AS BINARY(6)) AS UNIQUEIDENTIFIER)`, see the [article](https://www.informit.com/articles/article.aspx?p=25862&seqNum=7) about this. Your change break this and will cause issues to applications expecting NHibernate Guid.Comb to be equivalent to this SQL computing.